plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java

```diff
@@ -67,19 +72,43 @@ public BackpressureRestClient(RestClient delegate, ElasticsearchConfig config, T
 
     public void setHosts(HttpHost... hosts)
     {
-        delegate.setHosts(hosts);
+        delegate.setNodes(stream(hosts).map(Node::new).collect(toImmutableList()));
     }
 
     public Response performRequest(String method, String endpoint, Header... headers)
             throws IOException
     {
-        return executeWithRetries(() -> delegate.performRequest(method, endpoint, headers));
+        return executeWithRetries(() -> delegate.performRequest(toRequest(method, endpoint, headers)));
     }
 
     public Response performRequest(String method, String endpoint, Map<String, String> params, HttpEntity entity, Header... headers)
             throws IOException
     {
-        return executeWithRetries(() -> delegate.performRequest(method, endpoint, params, entity, headers));
+        return executeWithRetries(() -> delegate.performRequest(toRequest(method, endpoint, params, entity, headers)));
+    }
+
+    private static Request toRequest(String method, String endpoint, Header... headers)
```

Comment:
Can we move this method - below the `private static Request toRequest(String method, String endpoint, Map<String, String> params, HttpEntity entity, Header... headers)`